File: database/foxx/backend/src/model/documents/Comment.ts
import { Attribute, Document, Entity, Index } from 'type-arango';

export enum CommentSource {
    GOOGLE_PLACES = 'GOOGLE_PLACES',
    TRIPADVISOR = 'TRIPADVISOR',
}

// TODO: Implement bucket pattern (collect comments in single document per cenote)
@Document()
export class Comment extends Entity {
    // TODO: Change to cenoteKey
    @Index('hash')
    @Attribute()
    cenote_id: string;

    @Attribute()
    rating: number;

    @Attribute()
    source: CommentSource;

    @Attribute()
    text: string;

    @Attribute()
    timestamp: Date;

    @Attribute()
    url: string;
}

File: backend/src/main/java/org/cenoteando/models/Comment.java
package org.cenoteando.models;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
public class Comment {

    private int rating;
    private String text;
    private Date timestamp;

    public Comment(int rating, String text, Date timestamp) {
        this.rating = rating;
        this.text = text;
        this.timestamp = timestamp;
    }

}

